readthedocs/core/utils/__init__.py

```diff
@@ -184,6 +184,7 @@ def prepare_build(
             project=project,
             version=version,
             state=BUILD_STATE_TRIGGERED,
+            date__gte=timezone.now() - datetime.timedelta(minutes=5),
```

Comment:
This could use a comment explaining it, or pointing to this PR.

Seems we could adjust this to be the normal build time length, but that would add some queries.

Reply:
+1 on the comment.

> Seems we could adjust this to be the normal build time length, but that would add some queries.

I thought about this, but using the build time of the project, you end up waiting 15 or more before being able to trigger a new build when there is a staled build. In the end, it's not too much different than our task [`finish_inactive_builds`](https://github.com/readthedocs/readthedocs.org/blob/0da971556e9b8860230d601d74354548a4a6dde7/readthedocs/projects/tasks.py#L1819).

Oh, looking at that task it seems we changed the max time for a build to be considered stale to 2hs! It was `DOCKER_LIMIT['time'] * 1.2` originally. I think we can revert this changes now that we have the right docker limits in the dictionary: https://github.com/readthedocs/readthedocs.org/pull/7029

Reply:
`finish_inactive_builds` doesn't stop the builds from happening on the builders, it just updates the status of the Build object. They are solving different problems.

Reply:
Right, but `finish_inactive_builds` is the task that "unblocks" the DEDUPLICATE_BUILDS feature flag to start working again (not marking all of them as duplicated) by updating the state to something different than `triggered`. Now, it's unblocking these builds after more than 2hs of stale.

Reply:
Ah, I see. That makes sense.